magellan-library/src/test/java/com/wealthfront/magellan/view/LifecycleViewTest.kt

```diff
@@ -0,0 +1,39 @@
+package com.wealthfront.magellan.view
+
+import android.content.Context
+import android.widget.FrameLayout
+import org.junit.Before
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations.initMocks
+
+class LifecycleViewTest {
+
+  private lateinit var lifecycleView: LifecycleView<FrameLayout>
+
+  @Mock lateinit var frameLayout: FrameLayout
+  @Mock lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    initMocks(this)
+    lifecycleView = LifecycleView { frameLayout }
+  }
+
+  @Test
+  fun onCreateView() {
+    lifecycleView.show(context)
+
+    assertThat(lifecycleView.view).isEqualTo(frameLayout)
+  }
+
+  @Test
+  fun onDestroyView() {
+    lifecycleView.hide(context)
+
+    assertThat(lifecycleView.view).isEqualTo(null)
+  }
```

Comment:
We might want to make this test go through the whole lifecycle, declaring that the view either exists or doesn't along the way. This test doesn't test that `hide()` actually nulls the view out, for example.

Reply:
Added the `wholeLifecycle` testcase to check the state of the view in all the lifecycle callbacks.